cli/src/cli_util.rs

```diff
@@ -856,6 +857,18 @@ impl WorkspaceCommandHelper {
         Ok(FilesetExpression::union_all(expressions))
     }
 
+    pub fn auto_tracking_matcher(&self) -> Result<Box<dyn Matcher>, CommandError> {
+        let pattern = self.settings.config().get_string("snapshot.auto-track")?;
+        // This special case is to support users who have disabled filesets
+        // TODO: once we delete allow-filesets, we can default to "all()" in misc.toml
+        // and drop this special case
+        if pattern.is_empty() {
+            return Ok(Box::new(EverythingMatcher));
+        }
+        let expression = self.parse_file_patterns(&[pattern])?;
```

Comment:
nit: I think it's okay to disregard `allow-filesets` here. It's not path arguments, and `fix` patterns are parsed as filesets for example.

Reply:
This was specifically to support the default value from `misc.toml`. I've currently made that an empty string. If we make it `all()` instead, then we're going to break users who disable filesets. Setting it `.` doesn't work either, because it's relative to CWD by default, right?

Reply:
`ui.allow-filesets` controls whether command arguments are parsed as filesets (mainly for backward compatibility.) I don't think config values would have to respect this flag.

And, we'll have to use low-level `fileset::parse()` function to not resolve the auto-track patterns relative to cwd.